### PR TITLE
Added Moyal Distribution

### PR DESCRIPTION
**Pull request by Smit-create**
Added Moyal Distribution
#### References to other Issues or PRs

#14881

#### Release Notes

* stats
  * Added moyal distribution

### CODE REVIEW

**Comment by ethankward** [comment]
Tests can be simplified after #18451.

File: sympy/stats/crv_types.py [hunk 1]
```diff
@@ -2525,6 +2527,86 @@ def Maxwell(name, a):
...
+
+    def pdf(self, x):
+        mu, sigma = self.mu, self.sigma
+        num = exp(-(exp(-(x - mu)/sigma) + (x - mu)/(sigma))/2)
+        den = (sqrt(2*pi) * sigma)
+        return num/den
+
+    def _cdf(self, x):
+        mu, sigma = self.mu, self.sigma
+        return erfc(exp(-(x - mu)/(2*sigma))/sqrt(2))
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py]: This can be removed.

Comment (Smit-create) [reply sympy/stats/crv_types.py]: After removing `cdf` takes infinite time to evaluate.

Comment (czgdp1807) [reply sympy/stats/crv_types.py]: Removing only cdf doesnt affect the timings much. Therefore, only this can be removed and others have to be kept.

Comment (Smit-create) [reply sympy/stats/crv_types.py]: I removed only `_cdf(self, x)` and checked but it still takes infinite time to evaluate.

Comment (czgdp1807) [reply sympy/stats/crv_types.py]: On my system it works fine,
```python
>>> from sympy.stats import Moyal, cdf
>>> mu = Symbol("mu", real=True)
>>> sigma = Symbol("sigma", positive=True, real=True)
>>> X = Moyal("X", mu, sigma)
>>> z = Symbol("z")
>>> cdf(X)(z)
   ⎛                  ⎛     μ   -z ⎞               ⎞     
   ⎜           -μ     ⎜    ───  ───⎟            -μ ⎟   μ 
   ⎜           ───    ⎜    2⋅σ  2⋅σ⎟            ───⎟  ───
   ⎜           2⋅σ    ⎜√2⋅ℯ   ⋅ℯ   ⎟            2⋅σ⎟  2⋅σ
√2⋅⎜- √2⋅√π⋅σ⋅ℯ   ⋅erf⎜────────────⎟ + √2⋅√π⋅σ⋅ℯ   ⎟⋅ℯ   
   ⎝                  ⎝     2      ⎠               ⎠     
─────────────────────────────────────────────────────────
                          2⋅√π⋅σ                         
>>> simplify(_)
       ⎛    μ - z⎞
       ⎜    ─────⎟
       ⎜     2⋅σ ⎟
       ⎜√2⋅ℯ     ⎟
1 - erf⎜─────────⎟
       ⎝    2    ⎠
```

Comment (Smit-create) [reply sympy/stats/crv_types.py]: I just forgot to update my local environment. Yes, it worked. Thanks :)

File: sympy/stats/tests/test_continuous_rv.py [hunk 2]
```diff
@@ -848,6 +848,31 @@ def test_maxwell():
...
+                        *gamma(-I*sigma*z + Rational(1, 2))/sqrt(pi)
+    assert E(M) == mu + EulerGamma*sigma + sigma*log(2)
+    assert moment_generating_function(M)(z) == 2**(-sigma*z)*exp(mu*z) \
+                        *gamma(-sigma*z + Rational(1, 2))/sqrt(pi)
+
+    M = Moyal('M', 1, 2)
+    assert E(M) == 1 + 2*EulerGamma + 2*log(2)
+    assert variance(M).simplify() == 2*pi**2
+    assert kurtosis(M).simplify() == 7
+    assert skewness(M).simplify() == 28*sqrt(2)*zeta(3)/pi**3
```
Comment (czgdp1807) [inline sympy/stats/tests/test_continuous_rv.py]: IMO, these can be removed as symbolic tests are already there.

Comment (czgdp1807) [reply sympy/stats/tests/test_continuous_rv.py]: Are they affecting coverage in any way?

**Comment by czgdp1807** [comment]
Diff coverage is low, can you see which coverable lines are missed?

**Comment by Smit-create** [comment]
Screenshot from 2020-01-27 15-19-30

I am getting 100% diff coverage after removing numeric tests

File: sympy/stats/tests/test_continuous_rv.py (hunk 2 above)
Comment (Smit-create) [reply sympy/stats/tests/test_continuous_rv.py]: No, they are not affecting the coverage.

**Comment by Smit-create** [comment]
@czgdp1807  Please review. I don't know how diff coverage is changing, running locally it shows that I covered everything in the diff, also with the first commit the bot showed 100% diff coverage and then it got decreased surprisingly.

**Comment by czgdp1807** [comment]
It looks like an issue with Codecov, it is using incorrect diffs. LGTM. Will merge if no one objects for some time.

**Pull request merged**